File: lib/common/base/base.entity.ts
import { Column } from "typeorm";

export class BaseEntity {
  @Column({
    type: "bigint",
    nullable: true,
    name: "created_at"
  })
  createdAt?: number;

  @Column({
    type: "bigint",
    nullable: true,
    name: "updated_at"
  })
  updatedAt?: number;
}

File: server/traceo-common/traceo-common-jpa/src/main/java/org/traceo/common/jpa/base/BaseEntity.java
package org.traceo.common.jpa.base;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.id.uuid.UuidGenerator;

import java.io.Serializable;
import java.time.LocalDateTime;

@Getter @Setter
@MappedSuperclass
@Inheritance(strategy= InheritanceType.TABLE_PER_CLASS)
public class BaseEntity implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id", nullable = false, updatable = false)
    private String id;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updateAt;
}
